File: frontend/console/src/api/logic-area.js
import Axios from "@/http";

export function createLogicArea(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/logic-area', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveLogicArea(logicAreaId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/logic-area/${ logicAreaId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateLogicArea(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/logic-area', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteLogicArea(logicAreaId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/logic-area/${ logicAreaId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteLogicAreaWithFlag(logicAreaId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/logic-area/${ logicAreaId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllLogicAreas() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/logic-area/all').then(res => {
            resolve(res)
        })
    })
}

export function queryLogicArea (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/logic-area/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/logic_area.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_LOGIC_AREA = "/config/logic-area"
const URL_ROUTER_LOGIC_AREA_BY_ID = "/config/logic-area/:logic-area-id"
const URL_ROUTER_LOGIC_AREA_QUERY = "/config/logic-area/query"
const URL_ROUTER_LOGIC_AREA_ALL = "/config/logic-area/all"

func LogicAreaRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_LOGIC_AREA, config.CreateLogicArea)
	r.GET(URL_ROUTER_LOGIC_AREA_BY_ID, config.RetrieveLogicArea)
	r.PUT(URL_ROUTER_LOGIC_AREA, config.UpdateLogicArea)
	r.DELETE(URL_ROUTER_LOGIC_AREA_BY_ID, config.DeleteLogicArea)
	r.POST(URL_ROUTER_LOGIC_AREA_QUERY, config.QueryLogicArea)
	r.GET(URL_ROUTER_LOGIC_AREA_ALL, config.GetAllLogicAreas)
	return r
}